packages/core/src/render3/node_manipulation_i18n.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {assertDomNode, assertIndexInRange} from '../util/assert';
+import {TNode, TNodeType} from './interfaces/node';
+import {isProceduralRenderer, RElement, Renderer3, RNode} from './interfaces/renderer';
+import {LView} from './interfaces/view';
+import {getInsertInFrontOfRNodeWithNoI18n, nativeInsertBefore} from './node_manipulation';
+import {unwrapRNode} from './util/view_utils';
+
+
+/**
+ * Find a node in front of which `currentTNode` should be inserted (takes i18n into account).
+ *
+ * This method determines the `RNode` in front of which we should insert the `currentRNode`. This
+ * takes `TNode.insertBeforeIndex` into account.
+ *
+ * @param parentTNode parent `TNode`
+ * @param currentTNode current `TNode` (The node which we would like to insert into the DOM)
+ * @param lView current `LView`
+ */
+export function getInsertInFrontOfRNodeWithI18n(
+    parentTNode: TNode, currentTNode: TNode, lView: LView): RNode|null {
+  const tNodeInsertBeforeIndex = currentTNode.insertBeforeIndex;
+  const insertBeforeIndex =
+      Array.isArray(tNodeInsertBeforeIndex) ? tNodeInsertBeforeIndex[0] : tNodeInsertBeforeIndex;
+  if (insertBeforeIndex === null) {
+    return getInsertInFrontOfRNodeWithNoI18n(parentTNode, currentTNode, lView);
+  } else {
+    ngDevMode && assertIndexInRange(lView, insertBeforeIndex);
+    return unwrapRNode(lView[insertBeforeIndex]);
+  }
+}
+
+
+/**
+ * Process `TNode.insertBeforeIndex` by adding i18n text nodes.
+ *
+ * See `TNode.insertBeforeIndex`
+ *
+ * @param renderer
+ * @param childTNode
+ * @param lView
+ * @param childRNode
+ * @param parentRElement
+ * @param i18nChildren
+ */
+export function processI18nInsertBefore(
+    renderer: Renderer3, childTNode: TNode, lView: LView, childRNode: RNode|RNode[],
+    parentRElement: RElement|null): void {
+  const tNodeInsertBeforeIndex = childTNode.insertBeforeIndex;
+  if (Array.isArray(tNodeInsertBeforeIndex)) {
+    // An array indicates that there are i18n nodes that need to be added as children of this
+    // `rChildNode`. These i18n nodes were created before this `rChildNode` was available and so
+    // only now can be added. The first element of the array is the normal index where we should
+    // insert the `rChildNode`. Additional elements are the extra nodes to be added as children of
+    // `rChildNode`.
```

Comment:
rChildNode -> childRNode
```suggestion
    // `childRNode `. These i18n nodes were created before this `childRNode ` was available and so
    // only now can be added. The first element of the array is the normal index where we should
    // insert the `childRNode `. Additional elements are the extra nodes to be added as children of
    // `childRNode `.
```

Reply:
done